packages/engine/Source/DataSources/EntityCluster.js

```diff
@@ -869,6 +951,32 @@ function updateEnable(entityCluster) {
   disableCollectionClustering(entityCluster._labelCollection);
   disableCollectionClustering(entityCluster._billboardCollection);
   disableCollectionClustering(entityCluster._pointCollection);
+
+  if (allVisibleEntities.length > 0) {
+    const uniqueEntities = [...new Set(allVisibleEntities)];
```

Comment:
Don't use ([push.apply](https://github.com/CesiumGS/cesium/issues/12053) or) the `...spread` operator for potentially large collections. This should be
`const uniqueEntities = Array.from(new Set(allVisibleEntities));`

Don't drain performance with useless computations. When there are no listeners attached to that event, then there is no reason to compute all this. Insert some
`if (entityCluster._declusteredEvent.numberOfListeners !== 0) { ... }`
to compute this only when there are listeners registered.